core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemActor.java

```diff
@@ -141,42 +142,26 @@ public void configure(FS fileSystem, ParameterList parameterList, INamedObject o
 		this.owner=owner;
 		this.fileSystem=fileSystem;
 		this.parameterList=parameterList;
-		if (fileSystem instanceof IWritableFileSystem) {
-			actions.addAll(Arrays.asList(ACTIONS_WRITABLE_FS));
-		}
 
-		if (getAction() == null)
-			throw new ConfigurationException(owner.getClass().getSimpleName()+" ["+owner.getName()+"]: action must be specified");
-		if (!actions.contains(getAction()))
-			throw new ConfigurationException(owner.getClass().getSimpleName()+" ["+owner.getName()+"]: unknown or invalid action [" + getAction() + "] supported actions are " + actions.toString() + "");
-
-		if (getAction().equals(ACTION_READ2)) {
-			ConfigurationWarnings.add(owner, log, "action ["+ACTION_READ2+"] has been replaced with ["+ACTION_READ1+"]");
-			setAction(ACTION_READ1);
-		}
-		if (getAction().equals(ACTION_WRITE2)) {
-			ConfigurationWarnings.add(owner, log, "action ["+ACTION_WRITE2+"] has been replaced with ["+ACTION_WRITE1+"]");
-			setAction(ACTION_WRITE1);
-		}
-		
-		if (StringUtils.isNotEmpty(getBase64()) && !(getBase64().equals(BASE64_ENCODE) || getBase64().equals(BASE64_DECODE))) {
-			throw new ConfigurationException("attribute 'base64' can have value '"+BASE64_ENCODE+"' or '"+BASE64_DECODE+"' or can be left empty");
-		}
-		
 		if (parameterList!=null && parameterList.findParameter(PARAMETER_CONTENTS2) != null && parameterList.findParameter(PARAMETER_CONTENTS1) == null) {
 			ConfigurationWarnings.add(owner, log, "parameter ["+PARAMETER_CONTENTS2+"] has been replaced with ["+PARAMETER_CONTENTS1+"]");
 			parameterList.findParameter(PARAMETER_CONTENTS2).setName(PARAMETER_CONTENTS1);
 		}
-		
-		//Check if necessarily parameters are available
-		actionRequiresAtLeastOneOfTwoParametersOrAttribute(owner, parameterList, ACTION_WRITE1, PARAMETER_CONTENTS1, PARAMETER_FILENAME, "filename", getFilename());
-		actionRequiresAtLeastOneOfTwoParametersOrAttribute(owner, parameterList, ACTION_MOVE,   PARAMETER_DESTINATION, null, "destination", getDestination());
-		actionRequiresAtLeastOneOfTwoParametersOrAttribute(owner, parameterList, ACTION_COPY,   PARAMETER_DESTINATION, null, "destination", getDestination());
-		actionRequiresAtLeastOneOfTwoParametersOrAttribute(owner, parameterList, ACTION_RENAME, PARAMETER_DESTINATION, null, "destination", getDestination());
-		
+
+		if (StringUtils.isNotEmpty(getAction())) {
+			checkConfiguration(getAction());
+		} else if (parameterList == null || parameterList.findParameter(PARAMETER_ACTION) == null) {
+			throw new ConfigurationException(owner.getClass().getSimpleName()+" ["+owner.getName()+"]: either action or parameter ["+PARAMETER_ACTION+"] must be specified");
+		}
```

Comment:
I think it will be more clear to say: "either *attribute* [action] or parameter [action] must be specified"